### DISCUSSION TITLE
Default config files and python environment paths in METplus shared install

### DISCUSSION THREAD

**Discussion by jinfanti** [dtcenter/METplus#2102] [Installation] [answered]
Hello! We are working toward a shared install of MET/METplus at CPC. For the initial install, we compiled/etc on a local system (jhicks in the examples below), and then copied the install over to a central location. We've noticed that some of the paths when running METplus point to the local system instead of the paths to the shared location, and we are unsure how to change them, or if we need to re-install/compile in the central location. Namely, we find that when running METplus (1) the default conf files and (2) python env (despite activating one locally) come up as the local path.  

For example:
```
(1) DEBUG 1: Default Config File: **/cpc/home/jhicks/**MET11/MET-11.0.0-beta3/share/met/config/GridStatConfig_default
DEBUG 1: User Config File: /cpc/met/MET11/METplus-5.0.0/parm/met_config/GridStatConfig_wrapped
```
Where the GridStatConfig_default should ideally point to /cpc/met/ where our shared install is; and

```
(2) DEBUG 1: User Config File: /cpc/met/MET11/METplus-5.0.0/parm/met_config/GridStatConfig_wrapped
Traceback (most recent call last):
  File "/cpc/home/mcharles/pycharm-projects/met-poe-verif/prepare_data.py", line 14, in <module>
    met_data = xr.open_dataset('/cpc/data/forecasts/land_air/all_ranges/global/post-processed-model-fcsts/gefs/2023/03/23/[gefs-00z_rfcst-cal_tmean_20230323_week2.nc](http://gefs-00z_rfcst-cal_tmean_20230323_week2.nc/)')
  File "**/export/cpc-lw7-jhicks/jhicks/**.conda/envs/pyactivate/lib/python3.9/site-packages/xarray/backends/api.py", line 481, in open_dataset
    engine = plugins.guess_engine(filename_or_obj)
  File "**/export/cpc-lw7-jhicks/jhicks/**.conda/envs/pyactivate/lib/python3.9/site-packages/xarray/backends/plugins.py", line 148, in guess_engine
    raise ValueError(error_msg)
ValueError: found the following matches with the input file in xarray's IO backends: ['netcdf4', 'h5netcdf']. But their dependencies may not be installed, see [...]
```
Where the user activated their own conda env, but it seems to be defaulting to one the jhicks system. 

Do we need to do a fresh install on the shared drive to fix this, or is there a place to set these?

Thank you and let me know if I can provide additional details.

**Comment by j-opatz** [comment]
Hi Johnna,

And thank you for your question. I'm going to direct this to @jprestop, who has the most experience on our team with MET and METplus wrapers installations.

Julie, is there something that Johnna needs to set in the METplus wrappers or MET installation so it looks for the central installation, rather than local files?

**Comment by jprestop** [comment, marked as answer]
Hi @jinfanti.  We recommend that MET be installed in place and not moved.  However, if MET is compiled and installed in one place and the "share/met" directory is moved to some other location, the user would need to set the MET_BASE environment variable to point to that data.  So, the user should be able to set MET_BASE to the /new/path/to ... the directory containing "share/met".  For example:
```
export MET_BASE=/path_to/share/met
```
replacing "/path_to" with the path to MET's "share/met" directory.

Please have the user's try that and let us know how it goes.

**Reply by jprestop** [reply]
Hi @jinfanti.  I just wanted to follow up and see if you were able to try out having the users set MET_BASE and how that worked?

**Reply by jprestop** [reply]
Hi @jinfanti.  I just wanted to follow up and see if you were able to try out having the users set MET_BASE and how that worked?

**Reply by jprestop** [reply]
Hi @jinfanti.  I'm going to go ahead and mark this Discussion as answered and lock it.  Please follow up with a new Discussion if you experience any further problems.
